packages/core/src/actors.ts

```diff
@@ -128,12 +115,14 @@ export function fromCallback<TEvent extends EventObject>(
         if (isPromiseLike(dispose)) {
           dispose.then(
             (resolved) => {
-              sendNext(resolved);
-              sendComplete();
+              observers.forEach((o) => o.next?.(resolved));
```

Comment:
It IMHO would make sense to make `ObservableActorRef` responsible for emitting to observers. Ideally, the behavior would be responsible for just managing its own state and letting its direct owner (the `ObservableActorRef`) that something has changed. After all - one subscribes to an observable actor ref and not directly to behavior, so why behavior would be responsible for taking care of those multicast notifications?

Note that at the moment each behavior type has to implement this "broadcast" on its own, so there is a chance that a custom behavior could do it wrong etc, whereas in `ObservableActorRef` we can introduce a strong guarantee around this. 

With the proposed approach it will also be easier to fix the timing issue related to when observers are notified versus when the snapshot is being read by those subscribers.

At the moment we have a bug like this:
```
current = behavior.transition()
          -> observers.forEach(cb => cb(next))
            -> ref.getSnapshot() // oops, the `current` in the ObservableActorRef has not yet been updated
```

Whereas with this it wouldn't be an issue:
```
current = behavior.transition()
observers.forEach(cb => cb(next))
   -> ref.getSnapshot() // it's just fine as current has already been updated
```

Reply:
We have to think about what is "emittable". This is part of a bigger topic -  how do we keep state that should be private truly private if desired (not discoverable by anything)?

Reply:
In the spirit of keeping the discussion in public - we've quickly agreed on a call that we might think about "private" data for actors in the future. But right now even machines don't have such, so it's not worth worrying about.

I imagine that somebody could store such a private state in behaviors in a couple of ways - but there wouldn't be a clear strategy for persisting/rehydrating (and introspection) with any of those.